R/appshot.R

```diff
@@ -40,14 +42,73 @@ appshot.character <- function(app, file = "webshot.png", ...,
     p <- processx::process$new("R", args = c("--slave", "-e", cmd))
   })
 
+  # Make sure app is killed on exit
   on.exit({
     p$kill()
   })
 
   # Wait for app to start
   Sys.sleep(0.5)
 
+  # Get screenshot
   fileout <- webshot(sprintf("http://127.0.0.1:%d/", port), file = file, ...)
 
   invisible(fileout)
 }
+
+
+#' @rdname appshot
+#' @export
+appshot.shiny.appobj <- function(app, file = "webshot.png", ...,
+                              port = getOption("shiny.port"), envvars = NULL) {
+
+
+  port <- available_port(port)
+
+  args <- list(
+    url = sprintf("http://127.0.0.1:%d/", port),
+    file = file,
+    ...
+  )
+  r_session <- callr::r_bg(
+    function(...) {
+      # Wait for app to start
+      Sys.sleep(0.5)
```

Comment:
This delay should be a parameter. Same for `appshot.character`.

Reply:
I think that is fair for us to keep it internal.  It is a delay for webshot to start processing.  The delay in taking the picture could be increased by the 'delay' parameter. By keeping it static at 0.5, it is a minimum of 0.5 seconds in processing before the webshot is taken.